### PR TITLE
close #1442 create event ticket object

### LINKED ISSUE #1442: Create event ticket object

### PR DESCRIPTION
**Pull request by panhachom**
#### Resource 
https://developers.google.com/wallet/tickets/events/overview/how-classes-objects-work

image

photo1716949884

### CODE REVIEW

**Comment by panhachom** [comment]
####  Question  
 1. Pass Class Review Process 
 https://pay.google.com/business/console/passes/BCR2DN4TW6W4VZL2/issuer/3388000000022336356/eventticket/edit/3388000000022336356.tedx
 2. Add and delete call back 
  https://developers.google.com/wallet/tickets/events/use-cases/use-callbacks-for-saves-and-deletions
3. Google Pay development stage accessibility resources

**Comment by panhachom** [comment]
1. How Class review process work ? 
2. Pass Add and Callback ? 
   https://developers.google.com/wallet/tickets/events/use-cases/use-callbacks-for-saves-and-deletions
3. Google pay development stage accessibility resources

**Comment by panhachom** [comment]
- line-item format : L-line_item_id-OrderNumber (without R)
-  table use inheritance EventGoogleWalletClass(requirement base on event store in preferences ) < (GoogleWalletClass)
-  have middle table , product has one product_google_wallet and wallet has many product_google_wallet 
-  automate

File: app/models/spree_cm_commissioner/google_wallet.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module SpreeCmCommissioner
+  class GoogleWallet < SpreeCmCommissioner::Base
+    enum review_status: { DRAFT: 0, UNDER_REVIEW: 1, APPROVED: 2, REJECTED: 3 }
+
+    has_many :product_google_wallets, class_name: 'SpreeCmCommissioner::ProductGoogleWallet'
+    has_many :products, through: :product_google_wallets
+
+    has_one :logo, as: :viewable, dependent: :destroy, class_name: 'SpreeCmCommissioner::Asset'
```
Comment (theachoem) [inline app/models/spree_cm_commissioner/google_wallet.rb:8]:
```
def object_builder
   EventTicketObjectBuilder
end
```

result = line_item.google_wallet.object_builder.call(line_item)

**Review by theachoem — CHANGES_REQUESTED** [review]

**Pull request merged**